### ISSUE TITLE
Examples and/or documentation for running iOS tests on a physical device

### ISSUE THREAD

**Issue by jiawen** [bazelbuild/rules_apple#1215]
I've been using `rules_apple` successfully for a few projects and can run `ios_unit_test` in a simulator. However, there doesn't seem to be much documentation regarding how to run tests on a physical device. A great use case would be to easily measure on-device performance.

It would be tremendously helpful to have at least some documentation, if not working examples under the `//examples` directory.

I'm imagining something along the lines of:
* How to run a single `ios_unit_test` and `ios_ui_test`.
* How to run `ios_unit_test_suite` and `ios_ui_test_suite`.
* Examples of different `runner`s and `test_host`s and what they mean.
    * What's required in a host `ios_application`.
* Tulsi example.
* How this all fits with [google/xctestrunner](https://github.com/google/xctestrunner).

**Comment by keith** [comment]
I agree the general workflows here should be documented. But the tl;dr is that you need to use Tulsi (or some other Xcode bridge) to do anything on device. Bazel and the rules themselves have no knowledge of how to run on device. (xctestrunner does, but that isn't exposed through the rules today)

**Comment by jiawen** [comment]
Thanks! The pointer about needing Tulsi is already helpful!

I also discovered today via browsing through the issues that it's possible to get `bazel run` to execute tests inside a simulator with automated device matching - that's awesome! We should put it in the examples (I'm an Xoogler and happy to contribute).

**Comment by keith** [comment]
Great! We can definitely start some new document in docs/ with this kinda info

**Comment by thii** [comment]
@jiawen Can you try this patch https://github.com/bazelbuild/rules_apple/pull/1266?

**Comment by DoDoENT** [comment]
I did exactly the same and running test fails with

```
xctestrunner.shared.ios_errors.BundleError: Failed to extract signing identity from Executable
```

Running the very same test via Xcode generated by `rules_xcodeproj` works, so there is nothing wrong with my provisioning profile.
